src/extras/gizmo/gizmo.js

```diff
@@ -230,19 +235,38 @@ class Gizmo extends EventHandler {
     /**
      * Creates a new Gizmo object.
      *
-     * @param {AppBase} app - The application instance.
      * @param {CameraComponent} camera - The camera component.
-     * @param {Layer} layer - The render layer.
-     * @example
+     * @param {Layer} [layer] - The render layer. This can be provided by the user or will be created
+     * and added to the scene and camera if not provided. Successive gizmos will share the same layer
+     * and will be removed from the camera and scene when the last gizmo is destroyed.
      * const gizmo = new pc.Gizmo(app, camera, layer);
      */
-    constructor(app, camera, layer) {
+    constructor(camera, layer) {
+        Debug.assert(camera instanceof CameraComponent, 'Incorrect parameters for Gizmos\'s constructor. Use new Gizmo(camera, layer)');
         super();
 
-        this._app = app;
-        this._device = app.graphicsDevice;
         this._camera = camera;
-        this._layer = layer;
+        this._app = camera.system.app;
+        this._device = this._app.graphicsDevice;
+
+        if (layer) {
+            this._layer = layer;
+        } else {
+            if (!layerStore.has(this._app)) {
+                const gizmoLayer = new Layer({
+                    name: LAYER_NAME,
+                    clearDepthBuffer: true,
+                    opaqueSortMode: SORTMODE_NONE,
+                    transparentSortMode: SORTMODE_NONE
+                });
+                this._app.scene.layers.push(gizmoLayer);
+                this._camera.layers = this._camera.layers.concat(gizmoLayer.id);
+                layerStore.set(this._app, { layer: gizmoLayer, refs: 0 });
```

Comment:
the Map() stores reference to the App used as a key .. so unless it's removed from the map, it won't be garbage collected. Not ideal. Use DeviceCache which clears the device data when the device is destroyed.